docs/security/ldap-authentication.md

```diff
@@ -0,0 +1,288 @@
+    ```
+
+    To set and persist values at runtime, use the following statements:
+
+    ```sql
+    mysql> SET PERSIST authentication_ldap_simple_server_host='127.0.0.1';
+    mysql> SET PERSIST authentication_ldap_simple_bind_base_dn='dc=percona, dc=com';
+    ```
+
+=== "Load the SASL-based LDAP authentication plugin" 
+
+    ```sql
+    mysql> INSTALL PLUGIN authentication_ldap_sasl SONAME 'authentication_ldap_sasl.so`;
+    ```
+
+    To set and persist values at runtime, use the following statements:
+
+    ```sql
+    mysql> SET PERSIST authentication_ldap_sasl_server_host='127.0.0.1';
+    mysql> SET PERSIST authentication_ldap_sasl_bind_base_dn='dc=percona, dc=com';
+    ```
+
+## Create a user methods using simple LDAP authentication
+
+There are several methods to add or modify a user.
+
+=== "Use authentication_ldap_simple plugin"
+
+    In the `CREATE USER` statement or the `ALTER USER` statement, for simple LDAP authentication, you can specify the `authentication_ldap_simple` plugin in the `IDENTIFIED WITH` clause:
+
+    ```text
+    mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_simple;
+    ```
+
+    Using the `IDENTIFIED WITH` clause, the database server assigns the specified plugin.
+
+=== "Use the authentication string in simple LDAP"
+
+    If you provide the optional authentication string clause, ‘cn,ou,dc,dc’ in the example, the string is stored along with the password.
+
+    ```text
+    mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_simple BY 'cn=[user name],ou=[organization unit],dc=[domain component],dc=com'
+    ```
+
+    The following variables are not used when creating the user with the authentication string clause, for example, ‘cn,ou,dc,dc’:
+
+    * authentication_ldap_simple_bind_base_dn
+
+    * authentication_ldap_simple_bind_root_dn
+
+    * authentication_ldap_simple_bind_root_pwd
+
+    * authentication_ldap_simple_user_search_attr
+
+    * authentication_ldap_simple_group_search_attr
+
+    If you create the user with “IDENTIFIED BY authentication_ldap_simple” the listed variables are used.
+
+## Create a user methods using SASL-based LDAP authentication
+
+There are several methods to add or modify a user.
+
+=== "Use authentication_ldap_sasl plugin"
+
+    For SASL-based LDAP authentication, in the `CREATE USER` statement or the `ALTER USER` statement, you can specify the `authentication_ldap_sasl` plugin:
+
+    ```text
+    mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_sasl;
+    ```
+
+=== "Use the authentication string in SASL-based LDAP"
+
+    If you provide the optional authentication string clause, ‘cn,ou,dc,dc’ in the example, the string is stored along with the password.
+
+    ```text
+    mysql> CREATE USER ... IDENTIFIED WITH authentication_ldap_sasl BY 'cn=[user name],ou=[organization unit],dc=[domain component],dc=com'
+    ```
+
+    The following variables are not used when creating a user with an authentication string:
```

Comment:
bind_root_dn and bind_root_pwd is also used by the new group_mapping variable.

Reply:
Rewritten.